src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java

```diff
@@ -179,11 +216,15 @@ private static boolean isOnTokenWithAnnotation(DetailAST blockComment, int token
      * @return true if block comment is on specified token without modifiers
      */
     private static boolean isOnPlainClassMember(DetailAST blockComment, int memberType) {
-        return blockComment.getParent() != null
-                && blockComment.getParent().getType() == TokenTypes.TYPE
-                && blockComment.getParent().getParent().getType() == memberType
+        DetailAST parent = blockComment.getParent();
+        while (parent != null && parent.getType() == TokenTypes.DOT) {
```

Comment:
please add comment here "// type could be in fully qualified form, so we go up to Type token"